dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/BulkDataRequestManager.kt

```diff
@@ -33,43 +42,185 @@ class BulkDataRequestManager(
+                companyName = companyMappingEntry.value.companyName,
+                framework = metaData.dataType.toString(),
+                reportingPeriod = metaData.reportingPeriod,
+                datasetId = metaData.dataId,
+                datasetUrl = metaData.url,
+            )
+        }
+    }
+
+    private fun getAcceptedRequestCombinations(
+        validRequestCombinations: List<DataMetaInformationRequest>,
+        existingDatasets: List<DataMetaInformation>,
+    ): List<ValidSingleDataRequest> {
+        val existingCriteriaSet =
+            existingDatasets
+                .map {
+                    Triple(it.companyId, it.dataType, it.reportingPeriod)
+                }.toSet()
+
+        val nonExistingRequests =
+            validRequestCombinations
+                .filter { request ->
+                    val requestCriteria = Triple(request.companyId, request.dataType, request.reportingPeriod)
+
+                    require(!(request.companyId == null || request.dataType == null || request.reportingPeriod == null)) {
+                        "Request cannot have null values: $request"
+                    }
+                    requestCriteria !in existingCriteriaSet
+                }.map { request ->
+                    ValidSingleDataRequest(
+                        companyIdentifier = request.companyId!!,
+                        dataType = request.dataType!!,
+                        reportingPeriod = request.reportingPeriod!!,
+                    )
+                }
+
+        return nonExistingRequests.ifEmpty { emptyList() }
+    }
+
+    private fun generateRequestCombinations(
         dataTypes: Set<DataTypeEnum>,
         reportingPeriods: Set<String>,
-        datalandCompanyId: String,
-    ) {
-        for (framework in dataTypes) {
-            for (reportingPeriod in reportingPeriods) {
-                if (!utils.existsDataRequestWithNonFinalStatus(datalandCompanyId, framework, reportingPeriod)) {
-                    utils.storeDataRequestEntityAsOpen(datalandCompanyId, framework, reportingPeriod)
+        datalandCompanyIds: Set<String>,
+    ): List<DataMetaInformationRequest> =
+        datalandCompanyIds.flatMap { companyId ->
+            dataTypes.flatMap { dataType ->
+                reportingPeriods.map { period ->
+                    DataMetaInformationRequest(
+                        companyId = companyId,
+                        dataType = dataType,
+                        reportingPeriod = period,
+                        showOnlyActive = true,
+                        uploaderUserIds = null,
+                        qaStatus = null,
+                    )
                 }
             }
         }
+
+    private fun storeDataRequests(
+        validSingleDataRequests: List<ValidSingleDataRequest>,
+        userProvidedIdentifierToDatalandCompanyIdMapping: Map<String, CompanyIdAndName>,
+    ): Pair<List<DataRequestResponse>, List<DataRequestResponse>> {
```

Comment:
This function is rather confusing and should be split. Naming is also somewhat misleading. Lets talk about it.